From db536f5ed4825b7e56dea36e1aa4af7d840feb1d Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Tue, 13 Sep 2016 17:42:41 -0700
Subject: [PATCH 0525/1239] This patch adds support for setting the command
 prompt based on the board name.

I also reformatted some comments to make checkpatch happy.

Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
---
 board/cavium/thunderx/thunderx.c | 22 ++++++++++++++++---
 include/configs/thunderx_81xx.h  | 37 +++++++++++++++++++++-----------
 2 files changed, 44 insertions(+), 15 deletions(-)

diff --git a/board/cavium/thunderx/thunderx.c b/board/cavium/thunderx/thunderx.c
index 58821c6e7f..fa280dbea6 100644
--- a/board/cavium/thunderx/thunderx.c
+++ b/board/cavium/thunderx/thunderx.c
@@ -2,7 +2,7 @@
 /**
  * (C) Copyright 2014, Cavium Inc.
 **/
-
+#define DEBUG
 #include <common.h>
 #include <dm.h>
 #include <malloc.h>
@@ -57,6 +57,8 @@ DECLARE_GLOBAL_DATA_PTR;
 
 #define BOARD_TYPE "BOARD="
 
+char thunderx_prompt[CONFIG_THUNDERX_PROMPT_SIZE] = "ThunderX> ";
+
 int board_init(void)
 {
 	ulong fdt_addr = (ulong)gd->fdt_blob;
@@ -95,7 +97,7 @@ int dram_init(void)
 	return 0;
 }
 
-/*
+/**
  * Board specific reset that is system reset.
  */
 void reset_cpu(ulong addr)
@@ -115,16 +117,30 @@ bool alternate_pkg(void)
 	return (val >> 22) & 0x3;
 }
 
-/*
+/**
  * Board late initialization routine.
  */
 int board_late_init(void)
 {
+	const char *board;
 	debug("%s()\n", __func__);
 	thunderx_parse_bdk_config();
 
+	board = getenv("board");
 	printf("Board type: %s\n", getenv("board"));
 
+	if (board != NULL && !getenv("prompt")) {
+		snprintf(thunderx_prompt, sizeof(thunderx_prompt), "%s> ",
+			 board);
+		printf("Set prompt to \"%s\"\n", thunderx_prompt);
+		setenv("prompt", thunderx_prompt);
+	} else if (getenv("prompt")) {
+		printf("%s: prompt already set to \"%s\"\n",
+		       __func__, getenv("prompt"));
+	} else {
+		printf("%s: board is NULL\n", __func__);
+	}
+
 #ifdef DEBUG
 	dm_dump_all();
 #endif
diff --git a/include/configs/thunderx_81xx.h b/include/configs/thunderx_81xx.h
index fe16163ff4..958c172a3a 100644
--- a/include/configs/thunderx_81xx.h
+++ b/include/configs/thunderx_81xx.h
@@ -103,8 +103,8 @@
 /* Flat Device Tree Definitions */
 
 /**
- * The board code has additional modification that it wants to make to the flat
- * device tree before handing it off to the Linux kernel.
+ * The board code has additional modification that it wants to make to the
+ * flat device tree before handing it off to the Linux kernel.
  */
 #define CONFIG_OF_BOARD_SETUP
 
@@ -127,10 +127,11 @@
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + 64 * 1024 * 1024)
 
 /**
- * This is defined in the board config header.  This specified memory area will
- * get subtracted from the top (end) of memory and won't get "touched" at all
- * by U-Boot.  By fixing up gd->ram_size the Linux kernel should get passed the
- * now "corrected" memory size and won't touch it either.
+ * This is defined in the board config header.  This specified memory area
+ * will get subtracted from the top (end) of memory and won't get
+ * "touched" at all by U-Boot.  By fixing up gd->ram_size the Linux kernel
+ * should get passed the now "corrected" memory size and won't touch it
+ * either.
  */
 #define CONFIG_SYS_MEM_TOP_HIDE		0x1000000
 
@@ -165,8 +166,8 @@
 #endif
 
 /**
- * Only allow the Ethernet MAC address environment variable to be overwritten
- * once.
+ * Only allow the Ethernet MAC address environment variable to be
+ * overwritten once.
  */
 #define CONFIG_OVERWRITE_ETHADDR_ONCE
 
@@ -231,7 +232,10 @@
 /** Enable grepenv command */
 #define CONFIG_CMD_GREPENV
 
-/** Enable callback support when environment variables are set, or changed */
+/**
+ * Enable callback support when environment variables are set, or
+ * changed
+ */
 #define CONFIG_CMD_ENV_CALLBACK
 
 /** Enable date command */
@@ -282,7 +286,9 @@
 #define CONFIG_RTC_DS1337
 
 #define CONFIG_DDR_SPD
-#define CONFIG_SYS_SPD_ADDR_LIST {0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57}
+#define CONFIG_SYS_SPD_ADDR_LIST 	\
+	{ 0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57 }
+
 #define CONFIG_SYS_SPD_I2C_BUS 1
 
 #define CONFIG_CMD_SAVES
@@ -359,12 +365,19 @@
 /** Maximum size of the BDK flat device-tree */
 #define CONFIG_BDK_FDT_SIZE		0x20000
 
+#define CONFIG_THUNDERX_PROMPT_SIZE	32
+
 /* Monitor Command Prompt */
-#define CONFIG_SYS_CBSIZE		512	/** Console I/O Buffer Size */
+#define CONFIG_SYS_CBSIZE		1024	/** Console I/O Buffer Size */
 #define CONFIG_SYS_PBSIZE		(CONFIG_SYS_CBSIZE + \
-					 sizeof(CONFIG_SYS_PROMPT) + 16)
+					 CONFIG_THUNDERX_PROMPT_SIZE + 16)\
+
 #define CONFIG_SYS_PROMPT_HUSH_PS2	"> "
 #define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
+
+#undef CONFIG_SYS_PROMPT
+#define CONFIG_SYS_PROMPT		getenv("prompt")
+
 /** Enable long help support */
 #define CONFIG_SYS_LONGHELP
 /** Enable editing of the command line */
-- 
2.29.0

